build_runner_core/lib/src/asset/writer.dart

```diff
@@ -6,6 +6,7 @@ import 'dart:async';
 
 import 'package:build/build.dart';
 
+@deprecated
```

Comment:
Tis nice to give a reason `@Deprecated(...)`

Reply:
nothing is using it, and it's extremely unlikely anyone would use it. This shouldn't have been made public at all.

I'm only deprecating instead of deleting out of an abundance of caution, I don't think we really need a reason.